src/Qir/Runtime/unittests/QirRuntimeTests.cpp

```diff
@@ -750,8 +750,8 @@ TEST_CASE("Qubits: allocate, release, dump", "[qir_support]")
     REQUIRE(qs->count == 3);
     REQUIRE(qs->itemSizeInBytes == sizeof(void*));
 
-    Qubit last = *reinterpret_cast<Qubit*>(__quantum__rt__array_get_element_ptr_1d(qs, 2));
-    qstr       = __quantum__rt__qubit_to_string(last);
+    QUBIT* last = *reinterpret_cast<QUBIT**>(__quantum__rt__array_get_element_ptr_1d(qs, 2));
+    qstr        = __quantum__rt__qubit_to_string(last);
```

Comment:
For the future to keep in mind:
I got a feeling that there potentially can be different string representation of qubit ID generated 
* by the `QubitToString()` member functions (e.g. `FullStateSimulator::QubitToString()`)
* and the `__quantum__rt__qubit_to_string()` function. 

To avoid that ambiguity it probably makes sense for them all to call the same shared function to get a string representation of a qubit ID (such that they all generate the same string representation for the same qubit ID).

Reply:
Our runtime is already set up such that `__quantum__rt__qubit_to_string` will forward to `QubitToString` on whichever runtime driver is currently registered:
https://github.com/microsoft/qsharp-runtime/blob/06339b58e644c5559c8fb64ae98c28e90fdf8ffa/src/Qir/Runtime/lib/QIR/delegated.cpp#L150-L154
So those two functions will always produce the same result. However, there is no guarantee that two runtimes or simulators will produce the same result. For example, the full state simulator implements `QubitToString` like this:
https://github.com/microsoft/qsharp-runtime/blob/e61934496d3dbc7763e0013eb7ccda34322bde63/src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp#L233-L236
while the Toffoli simulator implements it like this:
https://github.com/microsoft/qsharp-runtime/blob/e61934496d3dbc7763e0013eb7ccda34322bde63/src/Qir/Runtime/lib/Simulators/ToffoliSimulator.cpp#L84-L87
so there is already a difference in our own simulators. But because each implementer of `IRuntimeDriver` is free to implement `QubitToString` however they want (and indeed every runtime is free to choose their own qubit representation) there is no guarantee that `__quantum__rt__qubit_to_string` will produce the same result across different runtimes or different simulators.